redis/redis.go

```diff
@@ -43,6 +43,9 @@ type Conn interface {
 
 	// Receive receives a single reply from the Redis server
 	Receive() (reply interface{}, err error)
+
+	// Peek returns true if these is a byte in the Recieve buffer (indicating a message awaits)
+	Peek() bool
```

Comment:
As discussed, this would break the API for all current consumers (outside of a version change/Go Modules).

If you were to take this route (not endorsing it either way), creating a `ConnPeeker` interface that embeds `Conn` would be the preferred way to go.

Reply:
> As discussed, this would break the API for all current consumers (outside of a version change/Go Modules).
> 
> If you were to take this route (not endorsing it either way), creating a `ConnPeeker` interface that embeds `Conn` would be the preferred way to go.

Good call, I agree with you.  I think I will abandon this direction and go with a secondary internal process.